DBQnA/docker_compose/intel/cpu/xeon/README.md

```diff
@@ -0,0 +1,178 @@
+
+### 2.1 Setup Environment Variables
+
+Since the `compose.yaml` will consume some environment variables, you need to setup them in advance as below.
+
+```bash
+# your_ip should be your external IP address, do not use localhost.
+export your_ip=$(hostname -I | awk '{print $1}')
+
+# Example: no_proxy="localhost,127.0.0.1,192.168.1.1"
+export no_proxy=${your_no_proxy},$your_ip
+
+# If you are in a proxy environment, also set the proxy-related environment variables:
+export http_proxy=${your_http_proxy}
+export https_proxy=${your_http_proxy}
+
+# Set other required variables
+
+export TGI_PORT=8008
+export TGI_LLM_ENDPOINT=http://${your_ip}:$TGI_PORT
+export HF_TOKEN=${HUGGINGFACEHUB_API_TOKEN}
+export LLM_MODEL_ID="mistralai/Mistral-7B-Instruct-v0.3"
+export POSTGRES_USER=postgres
+export POSTGRES_PASSWORD=testpwd
+export POSTGRES_DB=chinook
+export texttosql_port=9090
+```
+
+Note: Please replace with `your_ip` with your external IP address, do not use localhost.
+
+### 2.2 Start Microservice Docker Containers
+
+There are 2 options to start the microservice
+
+2.2.1 Start the microservice using docker compose
+
+```bash
+cd GenAIExamples/DBQnA/docker_compose/intel/cpu/xeon
+docker compose up -d
+```
+
+2.2.2 Alternatively we can start the microservices by running individual docker services
+
+**NOTE:** Make sure all the individual docker services are down before starting them.
+
+Below are the commands to start each of the docker service individually
+
+- Start PostgresDB Service
+
+We will use [Chinook](https://github.com/lerocha/chinook-database) sample database as a default to test the Text-to-SQL microservice. Chinook database is a sample database ideal for demos and testing ORM tools targeting single and multiple database servers.
+
+```bash
+
+docker run --name test-texttosql-postgres --ipc=host -e POSTGRES_USER=${POSTGRES_USER} -e POSTGRES_HOST_AUTH_METHOD=trust -e POSTGRES_DB=${POSTGRES_DB} -e POSTGRES_PASSWORD=${POSTGRES_PASSWORD} -p 5442:5432 -d -v $WORKPATH/comps/texttosql/langchain/chinook.sql:/docker-entrypoint-initdb.d/chinook.sql postgres:latest
+```
+
+- Start TGI Service
+
+```bash
+
+docker run -d --name="test-texttosql-tgi-endpoint" --ipc=host -p $TGI_PORT:80 -v ./data:/data --shm-size 1g -e HUGGINGFACEHUB_API_TOKEN=${HUGGINGFACEHUB_API_TOKEN} -e HF_TOKEN=${HF_TOKEN} -e model=${model} ghcr.io/huggingface/text-generation-inference:2.1.0 --model-id $model
+```
+
+```bash
+unset http_proxy
+
+docker run -d --name="test-texttosql-server" --ipc=host -p ${texttosql_port}:8090 --ipc=host -e http_proxy=$http_proxy -e https_proxy=$https_proxy -e TGI_LLM_ENDPOINT=$TGI_LLM_ENDPOINT opea/texttosql:latest
+```
+
+- Start React UI service
+
+```bash
+docker run -d --name="test-dbqna-react-ui-server" --ipc=host -p 5174:80 -e no_proxy=$no_proxy -e https_proxy=$https_proxy -e http_proxy=$http_proxy opea/dbqna-react-ui:latest
+```
+
+## 🚀 Validate Microservices
+
+3.1 TGI Service
```

Comment:
Please format the Heading  for section 3.1 to 3.3 including all the sub-headings